vllm/config/model.py

```diff
@@ -337,6 +337,22 @@ def compute_hash(self) -> str:
         assert_hashable(str_factors)
         return hashlib.sha256(str(factors).encode()).hexdigest()
 
+    def _apply_nested_overrides(
+            self, hf_config: "PretrainedConfig",
+            nested_overrides: dict[str, dict[str, Any]]) -> None:
+        for key, override in nested_overrides.items():
+            sub_config = getattr(hf_config, key, None)
+
+            if sub_config is None:
+                continue
+
+            if isinstance(sub_config, dict):
+                sub_config.update(override)
+                continue
+
+            for attr, value in override.items():
+                setattr(sub_config, attr, value)
+
```

Comment:
```suggestion
    def _update_nested(
        self,
        targets: "Union[PretrainedConfig, dict[str, Any]]",
        updates: dict[str, Any],
    ) -> None:
        for key, update in updates.items():
            if isinstance(updates, dict):
                _update_nested(
                    targets[key] if isinstance(targets, dict) else getattr(targets, key),
                    update,
                )
            
            if isinstance(targets, dict):
                targets[key] = update
            else:
                setattr(targets, key, update)
```

e.g. like this. Also we should add a test in `test_config.py`

Reply:
Good call, updated and added a unit test